File: src/test/java/org/algohub/engine/judge/FindClassNameSpec.java
package org.algohub.engine.judge;

import org.junit.Test;

import java.util.Optional;

import static org.assertj.core.api.Assertions.assertThat;
import static org.assertj.core.api.Assertions.fail;

public class FindClassNameSpec {

    private static final String JAVA_CODE = "public class MyClass { }";
    private static final String DUMMY_STRING = "abcdef {}";

    @Test
    public void findsClassInJavaCode() throws Exception {
        FindClassName findClassName = new FindClassName();

        Optional<String> result = findClassName.in(JAVA_CODE);

        if (result.isPresent()) {
            assertThat(result.get()).isEqualTo("MyClass");
        } else {
            fail("Class not found");
        }
    }

    @Test
    public void indicatesIfThereIsNoClassInTheInput() throws Exception {
        FindClassName findClassName = new FindClassName();

        Optional<String> result = findClassName.in(DUMMY_STRING);

        assertThat(result.isPresent()).isFalse();
    }
}

File: src/test/kotlin/org/algohub/engine/judge/FindClassNameSpec.kt
package org.algohub.engine.judge

import org.algohub.engine.compile.FindJavaClassName
import org.junit.Test

import java.util.Optional

import org.assertj.core.api.Assertions.assertThat
import org.assertj.core.api.Assertions.fail

class FindClassNameSpec {

    @Test
    @Throws(Exception::class)
    fun findsClassInJavaCode() {

        val result = FindJavaClassName().findIn(JAVA_CODE)

        if (result.isPresent) {
            assertThat(result.get()).isEqualTo("MyClass")
        } else {
            fail("Class not found")
        }
    }

    @Test
    @Throws(Exception::class)
    fun indicatesIfThereIsNoClassInTheInput() {

        val result = FindJavaClassName().findIn(DUMMY_STRING)

        assertThat(result.isPresent).isFalse()
    }

    companion object {
        private val JAVA_CODE = "public class MyClass { }"
        private val DUMMY_STRING = "abcdef {}"
    }
}